### PR TITLE
Marking service : store the marks applied for each answer

### PR DESCRIPTION
**Pull request by jon-shipley**
Add two new fields to the psychometric report:

* `QnSco` - the marks applied to the answer
* `QnTimeoutSco` - time out with correct answer (1= correct, 0= incorrect)

To do this we are storing all marked answers in the `answer` table which is produced from the marking service.

### CODE REVIEW

File: admin/bin/psychometricians-report-v2.js [hunk 1]
```diff
@@ -38,7 +40,13 @@ async function main () {
...
-  process.exit(0)
 }
 
 main()
+  .then(() => {
+    poolService.drain()
+  })
+  .catch(e => {
+    console.warn(e)
+    poolService.drain()
```
Comment (GuyHarwood) [inline admin/bin/psychometricians-report-v2.js]: 💯

File: admin/services/data-access/sql.pool.service.js [hunk 2]
```diff
@@ -29,10 +29,10 @@ let pool = null
 
 const sqlPoolService = {}
 
-  /**
-   * Initialise the connection pool.  Called once per application instance
-   * @return {Promise}
-   */
+/**
+ * Initialise the connection pool.  Called once per application instance
+ * @return {Promise}
```
Comment (GeorgeChatzigiannis) [inline admin/services/data-access/sql.pool.service.js]: Is the return needed here?
this method doesn't seem like it's returning anything.

File: admin/services/data-access/sql.pool.service.js [hunk 3]
```diff
@@ -41,10 +41,10 @@ sqlPoolService.init = () => {
   })
 }
 
-  /**
-   * Get a connection from the pool.
-   * @return {Promise}
-   */
+/**
+ * Get a connection from the pool.
+ * @return {Promise}
```
Comment (GeorgeChatzigiannis) [inline admin/services/data-access/sql.pool.service.js]: Same here

File: admin/services/data-access/sql.pool.service.js [hunk 4]
```diff
@@ -60,4 +60,11 @@ sqlPoolService.getConnection = () => {
   })
 }
 
+/**
+ * Disconnect all pool connections
```
Comment (GeorgeChatzigiannis) [inline admin/services/data-access/sql.pool.service.js]: JSDOc return is missing here.

File: admin/services/data-access/sql.pool.service.js (hunk 2 above)
Comment (jon-shipley) [reply admin/services/data-access/sql.pool.service.js]: removed

File: admin/services/data-access/sql.pool.service.js (hunk 3 above)
Comment (GuyHarwood) [reply admin/services/data-access/sql.pool.service.js]: getConnection returns a promise, see the full code

Comment (jon-shipley) [reply admin/services/data-access/sql.pool.service.js]: it certainly does resolve `resolve(connection)`

Comment (GeorgeChatzigiannis) [reply admin/services/data-access/sql.pool.service.js]: ok github diff didn't help me on seeing this clearly.

**Pull request merged**